components/eam/chem_proc/procfiles/cam/mo_imp_sol_cache.F90

```diff
@@ -409,56 +409,61 @@ subroutine imp_sol( base_sol, reaction_rates, het_rates, extfrc, delt, &
 !-----------------------------------------------------------------------      
 !   	... Non-convergence
 !-----------------------------------------------------------------------      
-		  fail_cnt = fail_cnt + 1
-		  nstep = get_nstep()
- 	          write(*,'('' imp_sol: Time step '',1p,e21.13,'' failed to converge @ (lchnk,lev,isec,nstep) = '',4i6)') dt,lchnk,lev,isec,nstep
-		  stp_con_cnt = 0
+                  fail_cnt = fail_cnt + 1
+                  nstep = get_nstep()
+                  if( do_diag ) then
```

Comment:
If we hide this message, we will never know if the solver has converged or not for all future chemistry packages. If we are fairly confident that convergence doesn't matter (which I think should be investigated further), we can make this change in the file under pp_{chemistry_package_name} folder. That way, it will be restricted to only one chemistry package (and hopefully it has been evaluated to be safe).

Reply:
Good point, I'll remove the if statement from this.

Reply:
Do we need to know if it fails to converge? It seems that the model is fine to carry on without convergence. If it's a critical thing to know then we can leave those message there. In some of my runs I was getting a lot of them, which was a bit annoying when I needed to scan the log files for the debugging info.

Reply:
I believe these messages appear only in v3. I don't remember seeing these in v1 or v2. @wlin7 and @tangq might know more about it. I believe it is safer to make these changes in `components/eam/src/chemistry/pp_superfast_mam5_resus_mom_soag/mo_imp_sol.F90` file. That way, it will impact just the v3 chemistry package not all the future chemistry packages. This file (`components/eam/chem_proc/procfiles/cam/mo_imp_sol_cache.F90`) is used by the chemistry preprocessor to create new F90 files for new chemistry packages. If we add this change here, all future chemistry packages will have this if condition and we will never know if the equations converged or not.

@wlin7 and @tangq : Could you please confirm if `pp_superfast_mam5_resus_mom_soag/` is the right folder for the v3 chemistry?

Reply:
@singhbalwinder made a good point that we don't want to hide all of these messages. On the other hand, writing out warnings for each grid box could be too much. In addition, treating chemistry packages case-by-case is not ideal.

It seems to me that if we write out the regional (not global) sums of non-converge points when `do_diag = false`, it will substantially reduce the warning messages (so not filling up the log file) while providing a rough idea about how serious the non-converge issue is. Based on this information, users can then decide whether or not turning on `do_diag`.

@whannah1 , how is `do_diag` specified? I assume it's false by default. How to change it to true?

Reply:
@singhbalwinder I talked with @whannah1 about this on slack, and this convergence message is actually not as much of an issue, because a failure to converge here just means that it has to retry with a smaller time step. It doesn't mean there is necessarily any problem with the solution, just that there's some extra computational cost associated with trying again. However, the other message below does mean that there is an inaccuracy in the solution and probably should not be suppressed.

Reply:
Ok, that makes sense to me. I just don't want to suppress messages that provide us with some insights. Sometimes these messages help us debug issues that can be hard to debug. I also see the point of getting too much noise in the log file. We can do what @tangq suggested and aggregate these messages and just display how many grid cells (if any) were affected by this non-convergence each time step or at a coarser time frequency.